From 44b1288f44fe37f5c2ddbf880ccc60a0e7e2738e Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 12 Sep 2018 11:02:11 +0300
Subject: [PATCH] version-hooks: check that the src dir is clean

Add a check for source directory cleanness.
Otherwise, we do not get any notification of uncommitted changes.

Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/package/version-hooks/version-hooks.mk b/package/version-hooks/version-hooks.mk
index 22987a7..5ffa47a 100644
--- a/package/version-hooks/version-hooks.mk
+++ b/package/version-hooks/version-hooks.mk
@@ -2,6 +2,9 @@
 	if [ -n $(SRCDIR) ]; then \
 		pushd $(SRCDIR); \
 		echo -n "-`git describe 2>/dev/null`" > $(@D)/localversion-habanalabs; \
+		if [ -n "`git status --porcelain 2>&1`" ]; then \
+			printf '%s' -dirty >> $(@D)/localversion-habanalabs; \
+		fi; \
 		popd; \
 	fi
 endef
